x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rule_tag_badge.tsx

```diff
@@ -16,55 +16,61 @@ const tagTitle = i18n.translate(
   }
 );
 
+export type RuleTagBadgeOptions = 'all' | 'default';
+
+interface RuleTagBadgeBasicOptions {
+  isOpen: boolean;
+  onClick: React.MouseEventHandler<HTMLButtonElement>;
+  onClose: () => void;
+}
+
 export interface RuleTagBadgeCommonProps {
+  showAllTags?: boolean;
   tags: string[];
   badgeDataTestSubj?: string;
   titleDataTestSubj?: string;
   tagItemDataTestSubj?: (tag: string) => string;
 }
-export type RuleTagBadgeConditionalProps =
-  | {
-      isOpen: boolean;
-      onClick: React.MouseEventHandler<HTMLButtonElement>;
-      onClose: () => void;
-    }
-  | {
-      spread: boolean;
-    };
-export type RuleTagBadgeProps = RuleTagBadgeCommonProps & RuleTagBadgeConditionalProps;
+
+export type RuleTagBadgeProps<T extends RuleTagBadgeOptions = 'default'> = T extends 'default'
```

Comment:
For me `RuleTagBadgeProps ` is bit complicated, with these types `RuleTagBadgeOptions`,  `RuleTagBadgeCommonProps`,  and `RuleTagBadgeBasicOptions`.